obd/elm327.py

```diff
@@ -290,12 +290,14 @@ def auto_baudrate(self):
             # so write a second one (again so that the lone CR doesn't repeat
             # the previous command)
             self.__port.write(b"\x7F\x7F\r\n")
+            self.__port.write(b"ATZ\r\n")
```

Comment:
Can you add a comment that this is for the STN11xx?

I'll test this on an ELM and see how it goes

Reply:
Yeah, absolutely.  But, I'll wait for your thoughts regarding the time.sleep() before I do anything else.

Reply:
Looking at the original code this is supposed to be a nonsense command. Why do we need to send an ATZ?

Reply:
Thanks for you questions. After reading the STN documentation, I think I know why it's not working. I believe I will have a new commit soon which will be much better.

Reply:
Hi, I just read both the ELM and STN documentation and they both appear to say that we _should be terminating our commands with carriage returns only_!

Is it possible that the implementation of python-OBD has been sending the wrong terminating characters all along?  What happens when you replace \r\n with \r for your ELM commands?  Remember, this requirement is for commands only.  Apparently, ELM responses can be \r or \r\n depending on (AT L1 or AT L0 commands) settings, also listed within the documentation, which we can discuss in more detail later.  But I can't find anywhere in the documentation where it says that you can or should terminate commands using \r\n.  It seems explicit that it should be carriage return only.

**Here's the documentation that I found for the STN chip and ELM chip.**

http://www.kds-online.com/Downloads/OBD/OBDLink_STN11xx-ds.pdf
https://www.elmelectronics.com/wp-content/uploads/2016/07/ELM327DS.pdf

**STN Chip Documentation**
"By default, responses from the STN11XX are terminated with a carriage return (0x0D). ATL1 command can be used to have the STN11XX append line feeds (0x0A) to the carriage returns." (Page 6 of 20)

**ELM Chip Documentation**
"Whether it is an ‘AT’ type internal command or a hex string for the OBD bus, all messages to the ELM327 must be terminated with a carriage return character (hex ‘0D’) before it will be acted upon." (Page 8 of 94)

"All of the responses from the ELM327 are terminated with a single carriage return character and, optionally, a linefeed character (depending on your settings)." (Page 8 of 94)

**Conclusion**
Therefore, STN chips don't work during auto-baud because it expect commands to be terminated with \r but the original code terminates commands with \r\n.  Why my code change above works, I have no idea.  But I just tested it without the line-feed and it worked without sleep or anything!

Also, if what I found is correct, the rest of python-OBD commands will need to be changed for both STN and ELM implementations.  Please let me know what you think.

Reply:
That is interesting. I haven't dug into it yet, but do you want to send a PR for changing the command termination and we can discuss it there? I will need to dedicate some time to read/test before we make a change that big.

Reply:
Yup, I will do just that!  Thanks :)

Reply:
Great! I'll close this PR. Just create a new one when you are ready.